src/sidechain.js

```diff
@@ -95,13 +95,17 @@ function deserializeScOutputs(buf: Buffer, offset: number) {
     const vBitVectorCertificateFieldConfig = [];
     let constant = '';
     let wCeasedVk = '';
+    const provingSystem = "CoboundaryMarlin";
```

Comment:
I'm discussing with the mainchain team currently if hard coding "CoboundaryMarlin" is the proper thing to do. From  PaoloT:
```
1. Mainchain currently supports both "CoboundaryMarlin" and "Darlin" proving systems, the type is read in the Sidechain Creation Output looking at the Certificate and CSW verification keys

2. SDK only supports CoboundaryMarlin, but it might start supporting also Darlin at some point the the future

3. In the meanwhile, theoretically speaking, someone could create a sidechain using Darlin without using the SDK
```

The issue here is that as there is no field in vsc_ccout that sets the proving system type, the information has do be inferred from the verification keys. That might require access to the zendoo cryptographic libraries which we don't (yet) have in Javascript.

If there is no quick/easy way to add support for detecting the proving system type I'd recommend to go with hard coded values for "CoboundaryMarlin" for now and add proper detection later.

Reply:
These are great finds, updated code to lookup proving system and conditionally add CSW data.